### PR TITLE
Login view pt. 2

### PR DESCRIPTION
**Pull request by vexonius**
| Portrait  | Landscape |
| ------------- | ------------- |
| Simulator Screen Shot - iPhone 13 Pro - 2022-07-15 at 10 53 34 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-15 at 10 53 36  |

### CODE REVIEW

**Review by marinzg — CHANGES_REQUESTED** [review]
There's no way to tap login on a smaller device in landscape mode 😅 

https://user-images.githubusercontent.com/9210723/179195068-09f642b2-6217-406a-b6b4-3afe16726240.mov

File: QuizApp/Presentation/Login/LoginViewController.swift [hunk 1]
```diff
@@ -31,6 +35,14 @@ class LoginViewController: BaseViewController {
...
 
         gradientLayer.frame = view.bounds
+        adaptComponentsForOrientationChanges()
+    }
+
+    private func adaptComponentsForOrientationChanges() {
+        let spacing = UIDevice.current.orientation.isLandscape ?
+            CustomConstants.usernameInputLandscapeOffset :
+            CustomConstants.usernameInputOffset
+        componentsStackView.setCustomSpacing(spacing, after: logoLabel)
```
Comment (marinzg) [inline QuizApp/Presentation/Login/LoginViewController.swift]: Try to avoid using device orientation in favour of size classes.

File: QuizApp/Presentation/Login/LoginViewController.swift [hunk 2]
```diff
@@ -66,29 +87,51 @@ extension LoginViewController: ConstructViewsProtocol {
...
+        componentsStackView.spacing = 18
+        componentsStackView.axis = .vertical
+        componentsStackView.alignment = .center
+        componentsStackView.distribution = .fill
     }
 
     func defineLayoutForViews() {
+        componentsStackView.snp.makeConstraints { make in
+            make.top.leading.trailing.equalTo(view.safeAreaLayoutGuide).inset(DesignConstants.Insets.componentsInset)
+            make.bottom.equalTo(view.safeAreaLayoutGuide.snp.bottom)
```
Comment (marinzg) [inline QuizApp/Presentation/Login/LoginViewController.swift]: I don't think you need to explicitly add `snp.bottom`. SnapKit automatically uses the edge that you're using.
```swift
make.top.equalTo(view.safeAreaLayoutGuide) // inferred .top
make.top.equalTo(view.safeAreaLayoutGuide.snp.bottom) // needs explicit .bottom
```

File: QuizApp/Presentation/Login/LoginViewController.swift [hunk 3]
```diff
@@ -66,29 +87,51 @@ extension LoginViewController: ConstructViewsProtocol {
...
+            make.centerX.equalToSuperview()
```
Comment (marinzg) [inline QuizApp/Presentation/Login/LoginViewController.swift:108]: Is this breaking constraints?

File: QuizApp/Presentation/Login/LoginViewController.swift [hunk 4]
```diff
@@ -66,29 +87,51 @@ extension LoginViewController: ConstructViewsProtocol {
...
+        componentsContainerView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview()
+            make.top.greaterThanOrEqualToSuperview()
+            make.bottom.lessThanOrEqualToSuperview()
+        }
+
         usernameInput.snp.makeConstraints { make in
-            make.leading.trailing.equalToSuperview().inset(DesignConstants.Insets.componentsInset)
+            make.top.equalTo(componentsContainerView)
+            make.width.equalToSuperview()
```
Comment (marinzg) [inline QuizApp/Presentation/Login/LoginViewController.swift:123]: Why did you move from edges to width?

File: QuizApp/Presentation/Login/LoginViewController.swift [hunk 5]
```diff
@@ -109,4 +152,10 @@ extension LoginViewController: UITextFieldDelegate {
         textField.styleForOutOfFocus()
     }
 
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+
+        return true
+    }
```
Comment (marinzg) [inline QuizApp/Presentation/Login/LoginViewController.swift:165]: Nice save.
In general, we use scroll view with IQKeyboardManagerSwift framework.

**Review by marinzg — CHANGES_REQUESTED** [review]

File: QuizApp/Presentation/Login/LoginViewController.swift (hunk 4 above)
Comment (vexonius) [reply QuizApp/Presentation/Login/LoginViewController.swift:123]: looks cleaner since the insets are put once on stackview

File: QuizApp/Presentation/Login/LoginViewController.swift (hunk 3 above)
Comment (vexonius) [reply QuizApp/Presentation/Login/LoginViewController.swift:108]: nope

**Review by marinzg — APPROVED** [review]
